xbmc/filesystem/SMBDirectory.cpp

```diff
@@ -122,6 +122,9 @@ bool CSMBDirectory::GetDirectory(const CURL& url, CFileItemList &items)
       if(StringUtils::EndsWith(strFile, "$") && aDir.type == SMBC_FILE_SHARE )
         continue;
 
+      if (StringUtils::StartsWith(strFile, "."))
+        hidden = true;
+
```

Comment:
BTW IMO this is a regression fix so I think you can merge asap.

As a further discussion perhaps we should do a regexp for hidden files we
could move out so other vfs could utilize? Not sure what other filename
based hidden there are, I know Mac has a few but perhaps they are all dot
prefixed?
Den 23 feb 2015 18:21 skrev "Matthias Kortstiege" <notifications@github.com
>:

> In xbmc/filesystem/SMBDirectory.cpp
> <https://github.com/xbmc/xbmc/pull/6529#discussion_r25182215>:
>
> > @@ -122,6 +122,9 @@ bool CSMBDirectory::GetDirectory(const CURL& url, CFileItemList &items)
> >        if(StringUtils::EndsWith(strFile, "$") && aDir.type == SMBC_FILE_SHARE )
> >          continue;
> >
> > +      if (StringUtils::StartsWith(strFile, "."))
> > +        hidden = true;
> > +
>
> @tamland <https://github.com/tamland> i prefer this one. It´s more
> consistent to the others.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/6529/files#r25182215>.
>